### DISCUSSION TITLE
Is there any ways to override securitySchemes in Swagger UI?

### DISCUSSION THREAD

**Discussion by kimminw00** [apache/airflow#36429] [Q&A] [answered]
When using Airflow REST API through Swagger UI, I would like to secure Airflow REST API only using JWT authentication and disable other securitySchemes(Basic, GoogleOpenId, Kerberos). Is there any ways to override [securitySchemes](https://swagger.io/docs/specification/authentication/oauth2) in Swagger UI?

**Comment by potiuk** [comment]
Follow https://airflow.apache.org/docs/apache-airflow/stable/security/api.html#roll-your-own-api-authentication

**Comment by kimminw00** [comment]
I implemented custom API authentication functions, but it seems that other functions need to be modified in order to extend Swagger UI.
https://github.com/apache/airflow/blob/78b2d82adf778443ac4a6efba0596acf473a70b3/airflow/www/extensions/init_views.py#L270-L286
From the above code, it looks like <code>v1.yaml</code> file must be modified to extend Swagger UI. From what I've found so far, there is no way to provide a customized <code>v1.yaml</code> file except building a custom airflow package. Is that correct?

**Reply by potiuk** [reply, marked as answer]
No idea to be honest - I am not sure what you are asking about. If you want to run API calls,backend should be enough. These are plain REST calls - and you can generate your own client (and for that yes, you need to generate your code based on modfied v1.yaml). When it comes to swagger UI - this one that is embedded in Airflow uses the built-in auth schemes, But Swagger UI is just. a demo /documentation and a way to explore the API - you absolutely do not need swagger UI to run your API calls.

**Reply by kimminw00** [reply]
I provide airflow services to users. 
So, airflow users find out which APIs can be used and how to authenticate from swagger UI.
However, even after I implemented JWT-based API authentication, 
it didn't show up in the "Available authorizations" window in swagger UI, 
leaving users in a situation where they did not know whether it had been actually implemented.

> When it comes to swagger UI - this one that is embedded in Airflow uses the built-in auth schemes, But Swagger UI is just. a demo /documentation and a way to explore the API

From the above, it seems there is no way to edit swagger UI.
Thank you for your help :)

---

Available authorizations window
![Available authorizations window](https://github.com/apache/airflow/assets/13493384/2b4f190b-305f-4bd8-9039-42e9137008ca)

**Reply by potiuk** [reply]
You can always attempt to contribute it back. I don't lknow how to do it but if you have a service that you charge money for, it makes sense to consider to contribute such a feature back to Airflow as a way to give back :)
